File: src/main/java/Color.java
public class Color
{
    public final static Color white = new Color(255, 255, 255);
    public final static Color black = new Color(0, 0, 0);
    public final static Color red = new Color(255, 0, 0);
    public final static Color green = new Color(0, 255, 0);
    public final static Color blue = new Color(0, 0, 255);

    private final int value;

    public Color(int value)
    {
        this.value = value;
    }

    public Color(int r, int g, int b, int a)
    {
        this(((a & 0xFF) << 24) |
                ((r & 0xFF) << 16) |
                ((g & 0xFF) << 8) |
                ((b & 0xFF) << 0));
    }

    public Color(int r, int g, int b)
    {
        this(r, g, b, 0);
    }

    public Color(double r, double g, double b)
    {
        this((int) (r * 255), (int) (g * 255), (int) (b * 255));
    }

    public int getValue()
    {
        return value;
    }
}

File: src/main/kotlin/utils/Color.kt
package utils

class Color(val value: Int) {
    val r: Int get() = value shr 16 and 0xFF
    val g: Int get() = value shr 8 and 0xFF
    val b: Int get() = value shr 0 and 0xFF

    constructor(r: Int, g: Int, b: Int, a: Int = 0) : this(
        a and 0xFF shl 24 or
                (r and 0xFF shl 16) or
                (g and 0xFF shl 8) or
                (b and 0xFF shl 0)
    )

    constructor(r: Double, g: Double, b: Double) : this((r * 255).toInt(), (g * 255).toInt(), (b * 255).toInt()) {}

    companion object {
        val WHITE = Color(255, 255, 255)
        val BLACK = Color(0, 0, 0)
        val RED = Color(255, 0, 0)
        val GREEN = Color(0, 255, 0)
        val BLUE = Color(0, 0, 255)
    }

    operator fun times(d: Double): Color = Color((r * d).toInt(), (g * d).toInt(), (b * d).toInt())
}